### ISSUE TITLE
Component summary view by type

### ISSUE THREAD

**Issue by bvaughn** [facebook/react#20881]
A DevTools user has requested the following feature (reworded by me):
Expose an ordered table of all component types (names) along with the number of instances currently mounted. This would help identify the most impactful components to examine for memory related optimizations.

The Profiler isn't very useful in this scenario because it's too CPU-centric.

cc @yungsters

**Comment by ChrisDobby** [comment]
Is this the kind of thing you were thinking of?

<img width="974" alt="Screenshot 2021-02-28 at 08 04 59" src="https://user-images.githubusercontent.com/434389/109414114-109b9e00-79a9-11eb-8891-fdfba0720874.png">

**Comment by bvaughn** [comment]
Hmm... I think I'm picturing more of a traditional table layout for this. I think a table layout would work better because we may add other info besides just name and count, but also because it would cause components to be clipped like this (e.g. `RequesterLin...`)

**Comment by ChrisDobby** [comment]
Something a bit more like this:

<img width="680" alt="Screenshot 2021-02-28 at 18 10 04" src="https://user-images.githubusercontent.com/434389/109491728-df42d100-7a81-11eb-88a7-c4e7bbd7a693.png">

**Comment by bvaughn** [comment]
Something more like that, yeah. At least in layout.

**Comment by windmaomao** [comment]
Yeah i'll vote up with the table view, one of the thing good about Table is that we can sort on the column. The time each component spent is currently displayed, maybe the number of times it's used for this render is also useful one. 

Right now the most info i can get is the color which indicates if it renders or bails out. But i sometimes wish it can tell me more, ex. excessive renders. And also i sometimes confused about the time collected, is that the total time spent in each render function. If not, then time aggregation on the component NAME might be also useful.
